library/mobjects.c

```diff
@@ -69,21 +69,23 @@ int eblob_base_setup_data(struct eblob_base_ctl *ctl, int force)
 	struct stat st;
 	int err;
 
-	err = fstat(ctl->index_fd, &st);
-	if (err) {
-		err = -errno;
-		goto err_out_exit;
+	if (ctl->index_ctl.fd >= 0) {
```

Comment:
That code looks strange, `eblob_base_setup_data()` should always return error in this case and do not even try to open index files.

You hide this either error code or some unclear logic, which should be documented instead.